compiler/rustc_typeck/src/check/writeback.rs

```diff
@@ -195,6 +195,48 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
         }
     }
 
+    // (ouz-a #100498): Normally `[T] : std::ops::Index<usize>` should be normalized
+    // into [T] but currently `Where` clause stops the normalization process for it,
+    // here we check if the expr is `LangItem::Index` and index_ty is `usize` if so we don't
+    // `fix_index_builtin_expr`.
+    fn is_builtin_index(
+        &mut self,
+        e: &hir::Expr<'_>,
+        base: Ty<'tcx>,
+        index: Ty<'tcx>,
+        base_span: Span,
+        index_span: Span,
+    ) -> bool {
+        let mut ret = false;
+        if (&base).builtin_index().is_some() {
```

Comment:
```suggestion
        if base.builtin_index().is_some() {
```

Also, we could invert this, like:

```
if base.builtin_index().is_none() {
  return false;
}
```

That saves us an indentation level.